cdc/capture/http_util.go

```diff
@@ -0,0 +1,243 @@
+	}
+
+	// set sortEngine and EnableOldValue
+	_, captureInfos, err := capture.etcdClient.GetCaptures(ctx)
+	if err != nil {
+		return nil, cerror.ErrPDEtcdAPIError.Wrap(err)
+	}
+	cdcClusterVer, err := version.GetTiCDCClusterVersion(captureInfos)
+	if err != nil {
+		return nil, err
+	}
+	sortEngine := model.SortUnified
+	if !cdcClusterVer.ShouldEnableOldValueByDefault() {
+		replicaConfig.EnableOldValue = false
+		log.Warn("The TiCDC cluster is built from unknown branch or less than 5.0.0-rc, the old-value are disabled by default.")
+		if !cdcClusterVer.ShouldEnableUnifiedSorterByDefault() {
+			sortEngine = model.SortInMemory
+		}
+	}
+
+	// init ChangefeedInfo
+	info := &model.ChangeFeedInfo{
+		SinkURI:           changefeedConfig.SinkURI,
+		Opts:              make(map[string]string),
+		CreateTime:        time.Now(),
+		StartTs:           changefeedConfig.StartTS,
+		TargetTs:          changefeedConfig.TargetTS,
+		Config:            replicaConfig,
+		Engine:            sortEngine,
+		State:             model.StateNormal,
+		SyncPointEnabled:  false,
+		SyncPointInterval: 10 * time.Minute,
+		CreatorVersion:    version.ReleaseVersion,
+	}
+
+	if !replicaConfig.ForceReplicate && !changefeedConfig.IgnoreIneligibleTable {
+		ineligibleTables, _, err := verifyTables(capture, replicaConfig, changefeedConfig.StartTS)
+		if err != nil {
+			return nil, err
+		}
+		if len(ineligibleTables) != 0 {
+			return nil, cerror.ErrTableIneligible.GenWithStackByArgs(ineligibleTables)
+		}
+	}
+
+	err = verifySink(ctx, info.SinkURI, info.Config, info.Opts)
+	if err != nil {
+		return nil, err
+	}
+
+	return info, nil
+}
+
+func verifyTables(capture *Capture, replicaConfig *config.ReplicaConfig, startTs uint64) (ineligibleTables, eligibleTables []model.TableName, err error) {
+	meta, err := kv.GetSnapshotMeta(capture.kvStorage, startTs)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	filter, err := filter.NewFilter(replicaConfig)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+
+	snap, err := entry.NewSingleSchemaSnapshotFromMeta(meta, startTs, false /* explicitTables */)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+
+	for tID, tableName := range snap.CloneTables() {
+		tableInfo, exist := snap.TableByID(tID)
+		if !exist {
+			return nil, nil, errors.NotFoundf("table %d", tID)
+		}
+		if filter.ShouldIgnoreTable(tableName.Schema, tableName.Table) {
+			continue
+		}
+		if !tableInfo.IsEligible(false /* forceReplicate */) {
+			ineligibleTables = append(ineligibleTables, tableName)
+		} else {
+			eligibleTables = append(eligibleTables, tableName)
+		}
+	}
+	return
+}
+
+func verifySink(ctx context.Context, sinkURI string, cfg *config.ReplicaConfig, opts map[string]string) error {
+	filter, err := filter.NewFilter(cfg)
+	if err != nil {
+		return err
+	}
+	errCh := make(chan error)
+	s, err := sink.NewSink(ctx, "cli-verify", sinkURI, filter, cfg, opts, errCh)
```

Comment:
this will new a sink and run to verify?

Reply:
Yes, I have not find a better alternative way.